### PR TITLE
Issues #49 / #50 : Fixes for "item links" parsed variables

### PR DESCRIPTION
**Pull request by Solanya**
Variables are now parsed in the item browser tooltips (see Create item > Create from....)
When using an item ID as a variable in item description / use text, the color is now properly reset to that field color.

**Note : the item browser tooltips don't reset properly as they use the TRP3 list tool. Unsure how to properly fix for them, it's less visible as it resets to a yellow color instead of orange.
You also stated in #50 you wanted to make a pass with Ellyb's colors, just reminding.**

### CODE REVIEW

File: totalRP3_Extended/inventory/container.lua [hunk 1]
```diff
@@ -204,7 +204,15 @@ local function showItemTooltip(frame, slotInfo, itemClass, forceAlt, anchor)
 	end
 
 	if text1 and text1:len() > 0 then
-		TRP3_ItemTooltip:AddLine(text1, 1, 1, 1,true);
+		TRP3_ItemTooltip:AddLine(text1, 1, 0.67, 0,true);		-- corresponds to color("o") = FFAA00 for the description
```
Comment (Ellypse) [inline totalRP3_Extended/inventory/container.lua]: Use Ellyb.ColorManager.Orange:GetRed() etc. instead (so the color is consistent and if we change it it changes everywhere)

File: totalRP3_Extended/inventory/container.lua [hunk 2]
```diff
@@ -204,7 +204,15 @@ local function showItemTooltip(frame, slotInfo, itemClass, forceAlt, anchor)
...
-		TRP3_ItemTooltip:AddLine(text1, 1, 1, 1,true);
+		TRP3_ItemTooltip:AddLine(text1, 1, 0.67, 0,true);		-- corresponds to color("o") = FFAA00 for the description
+		_G["TRP3_ItemTooltipTextLeft"..i]:SetFontObject(GameFontNormal);
+		_G["TRP3_ItemTooltipTextLeft"..i]:SetSpacing(2);
+		_G["TRP3_ItemTooltipTextLeft"..i]:SetNonSpaceWrap(true);
+		i = i + 1;
+	end
+
+	if text1_lower and text1_lower:len() > 0 then
+		TRP3_ItemTooltip:AddLine(text1_lower, 0, 1, 0,true);	-- corresponds to color("g") = 00FF00 for the use text
```
Comment (Ellypse) [inline totalRP3_Extended/inventory/container.lua]: Use Ellyb.ColorManager.GREEN:GetRed() etc. instead.

File: totalRP3_Extended_Tools/script/element/element.lua [hunk 3]
```diff
@@ -153,7 +153,8 @@ local function decorateBrowserLine(frame, index)
 			end
 		end
 		if base.DE then
-			text = text .. "\n|cffff9900\"" .. base.DE .. "\"";
+			local argsStructure = {object = {id = objectID}};
+			text = text .. "\n|cffff9900\"" .. TRP3_API.script.parseArgs(base.DE .. "\"", argsStructure);
```
Comment (Ellypse) [inline totalRP3_Extended_Tools/script/element/element.lua]: Use Ellyb.ColorManger.ORANGE(text) instead.

**Comment by Solanya** [comment]
Before pulling, I only changed what you asked but should we update all color usages in both files ?

**Comment by Ellypse** [comment]
If you want to do it sure. If not, it's not really necessary. I juste want the new code to use the new stuff, the old code will be modernized at some point anyway.

**Comment by Solanya** [comment]
You should probably check the new commit to be sure, there's still changes to be done in those files but honestly too much for now x_x

**Pull request merged**